src/test/deploy/functions/cel.spec.ts

```diff
@@ -0,0 +1,530 @@
+import { expect } from "chai";
+import { resolveExpression, ExprParseError } from "../../../deploy/functions/cel";
+import { ParamValue } from "../../../deploy/functions/params";
+
+function stringV(value: string): ParamValue {
+  return new ParamValue(value, false, { string: true, number: false, boolean: false });
+}
+function numberV(value: number): ParamValue {
+  return new ParamValue(value.toString(), false, { string: false, number: true, boolean: false });
+}
+function boolV(value: boolean): ParamValue {
+  return new ParamValue(value.toString(), false, { string: false, number: false, boolean: true });
+}
+
+describe("CEL evaluation", () => {
+  describe("Identity expressions", () => {
+    it("raises when the referenced parameter does not exist", () => {
+      expect(() => {
+        resolveExpression("number", "{{ params.FOO }}", {});
+      }).to.throw(ExprParseError);
+    });
+
+    it("raises when the referenced parameter is of the wrong type", () => {
+      expect(() => {
+        resolveExpression("number", "{{ params.FOO }}", {
+          FOO: boolV(true),
+        });
+      }).to.throw(ExprParseError);
+    });
+
+    it("pulls number parameters", () => {
+      const params: Record<string, ParamValue> = {
+        FOO: numberV(22),
+        BAR: boolV(true),
+        BAZ: new ParamValue("quox", false, { string: true }),
```

Comment:
nit* `stringV("quox")`? or is this intentional?

Reply:
Oops. I had them all typed out, decided it was a farce later, and did a find-replace with the helper functions. Must've missed this one.